examples/display_camera/main/display_camera.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-FileCopyrightText: 2022 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+#include <stdio.h>
+#include "sdkconfig.h"
+#include "bsp/esp32_s2_kaluga_kit.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "freertos/semphr.h"
+#include "driver/spi_master.h"
+#include "esp_err.h"
+#include "esp_log.h"
+#include "esp_lcd_panel_io.h"
+#include "esp_lcd_panel_vendor.h"
+#include "esp_lcd_panel_ops.h"
+#include "esp_camera.h"
+
+static const char *TAG = "example";
+
+void app_main(void)
+{
+    bsp_i2c_init();
+    bsp_display_start();
+
+    // Initialize the camera
+    const camera_config_t camera_config = BSP_CAMERA_DEFAULT_CONFIG;
+    esp_err_t err = esp_camera_init(&camera_config);
+    if (err != ESP_OK) {
+        ESP_LOGE(TAG, "Camera Init Failed");
+        return;
+    }
+    sensor_t *s = esp_camera_sensor_get();
+    s->set_vflip(s, 1);
+    ESP_LOGI(TAG, "Camera Init done");
+
+    // Create LVGL canvas for camera image
+    lv_obj_t *camera_canvas = lv_canvas_create(lv_scr_act());
+    assert(camera_canvas);
+    lv_obj_center(camera_canvas);
+
+    // Get first frame and initialize the canvas with it
+    camera_fb_t *pic = esp_camera_fb_get();
+    assert(pic->buf);
+    lv_canvas_set_buffer(camera_canvas, pic->buf, BSP_LCD_H_RES, BSP_LCD_V_RES, LV_IMG_CF_TRUE_COLOR);
+    esp_camera_fb_return(pic);
+
+    while (1) {
+        pic = esp_camera_fb_get();
```

Comment:
Must it be here? Must it be called in loop? If not, it is on the top. 
I would like see something like this, if is it possible:

```
while(1)
{
esp_camera_fb_return(pic);
ESP_LOGI(TAG, "picture: %d x %d %dbyte", pic->width, pic->height, pic->len);
lv_obj_invalidate(camera_canvas); // This will force rewrite of the canvas
}
```

If not possible, use my code and add the `esp_camera_fb_get `before `esp_camera_fb_return`. I don't know this camera API, but i looks better.

Reply:
Yes, esp32-camera reads frames from the camera on the background. `esp_camera_fb_get` just returns (of of possible many) filled frame buffer.

After we are done with the processing, it must be returned to esp32-camera with `esp_camera_fb_return`